pandas/core/dtypes/cast.py

```diff
@@ -366,6 +367,8 @@ def maybe_promote(dtype, fill_value=np.nan):
     elif is_bool(fill_value):
         if not issubclass(dtype.type, np.bool_):
             dtype = np.object_
+        else:
+            fill_value = np.bool_(fill_value)
```

Comment:
you added some code here but are these actually hit?

Reply:
i see you have another PR on top; can you just do them in one

Reply:
> you added some code here but are these actually hit?

Yes.  All the changes to the non-test code are needed to make the changes in the test code pass.

Reply:
> i see you have another PR on top; can you just do them in one

Sure.  Just close this and move conversation there?

Reply:
yeah let's just do it in one